### PR TITLE
Replacing querySelector with their angular equivalent

### PR DESCRIPTION
**Pull request by MarkvdVorst**
Closes issue [#409](https://github.com/wearefrank/ladybug-frontend/issues/409)

Changed the existing document.querySelector with the ViewChild elementRef.

### CODE REVIEW

File: src/app/shared/components/toast/toast.component.ts [hunk 1]
```diff
@@ -45,11 +47,9 @@ export class ToastComponent implements OnInit, OnDestroy {
   }
 
   copyToClipboard(): void {
-    const text = document.querySelector('#detailedErrorMessage')!;
-    navigator.clipboard.writeText(text.innerHTML).then(() => {
-      const button = document.querySelector('#CopyToClipboard')!;
-      button.innerHTML = 'Copied!';
-      button.setAttribute('style', 'background-color: #23c6c8 !important');
+    navigator.clipboard.writeText(this.detailedErrorMessageElement.nativeElement.innerHTML).then(() => {
```
Comment (Matthbo) [inline src/app/shared/components/toast/toast.component.ts:50]: Does this work on Firefox? I know Firefox doesnt have the new Clipboard API implemented yet and the old one is already deprecated.
If it works then fine, if it doesnt then use the deprecated clipboard way

File: src/app/test/test.component.ts [hunk 2]
```diff
@@ -53,6 +53,7 @@ export class TestComponent implements OnInit, AfterViewInit {
   @ViewChild(TestSettingsModalComponent) testSettingsModal!: TestSettingsModalComponent;
   @ViewChild(DeleteModalComponent) deleteModal!: DeleteModalComponent;
   @ViewChild(TestFolderTreeComponent) testFileTreeComponent!: TestFolderTreeComponent;
+  @ViewChild('moveToInput') moveToInput!: ElementRef<HTMLInputElement>;
```
Comment (MatthijsSmets) [inline src/app/test/test.component.ts:56]: Maybe change this to ngmodel

File: src/app/test/test.component.ts [hunk 3]
```diff
@@ -289,7 +290,7 @@ export class TestComponent implements OnInit, AfterViewInit {
   updatePath(action: string): void {
     let reportIds: string[] = this.helperService.getSelectedIds(this.reports);
     if (reportIds.length > 0) {
-      let path: string = (document.querySelector('#moveToInput')! as HTMLInputElement).value;
+      let path: string = this.moveToInput.nativeElement.value;
```
Comment (MatthijsSmets) [inline src/app/test/test.component.ts:293]: This should not be the way to get the value of an input field in angular, maybe use ngmodel?

File: src/app/report/edit-display/edit-display.component.ts [hunk 4]
```diff
@@ -80,12 +81,11 @@ export class EditDisplayComponent {
   rerunReport(): void {
     let reportId: string = this.report.storageId;
     this.httpService.runDisplayReport(reportId, this.currentView.storageName).subscribe((response) => {
-      let element = document.querySelector('#showRerunResult')!;
       if (this.report == response) {
-        element.setAttribute('style', 'background-color: green');
+        this.rerunResultElement.nativeElement.setAttribute('style', 'background-color: green');
```
Comment (MatthijsSmets) [inline src/app/report/edit-display/edit-display.component.ts:85]: I think it's better to use a class variable called runSuccessful or something and then use ngClass or style.background-color in the html to apply a different background color

**Review by MatthijsSmets — CHANGES_REQUESTED** [review]
Not exactly what I had in mind, please use Angular solutions for styling and input fields like I suggested

File: src/app/report/edit-display/edit-display.component.html [hunk 5]
```diff
@@ -38,7 +38,7 @@
   </div>
 </div>
 
-<p id="showRerunResult">
+<p id="showRerunResult" [style]="rerunBackgroundColor">
   {{ rerunResult }}
```
Comment (MatthijsSmets) [inline src/app/report/edit-display/edit-display.component.html:42]: However i would suggest leaving the color in the html and just using a boolean to determine which color to display:
```suggestion
<p id="showRerunResult" [ngStyle]="{'background-color': rerunSuccess ? 'green' : 'red'}">
  {{ rerunSuccess ? '[Rerun succeeded]' : '[Rerun failed]' }}
```

File: src/app/report/edit-display/edit-display.component.ts [hunk 6]
```diff
@@ -57,6 +57,7 @@ export class EditDisplayComponent {
   @ViewChild('variables') variables!: ElementRef;
   saveOrDiscardType: string = '';
   differenceModal: DifferenceModal[] = [];
+  rerunBackgroundColor: string = 'background-color: green';
```
Comment (MatthijsSmets) [inline src/app/report/edit-display/edit-display.component.ts:60]:
```suggestion
  rerunSuccess: boolean = true;
```

File: src/app/report/edit-display/edit-display.component.ts [hunk 7]
```diff
@@ -80,12 +81,11 @@ export class EditDisplayComponent {
...
-      let element = document.querySelector('#showRerunResult')!;
       if (this.report == response) {
-        element.setAttribute('style', 'background-color: green');
+        this.rerunBackgroundColor = 'background-color: green';
         this.rerunResult = '[Rerun succeeded]';
       } else {
-        element.setAttribute('style', 'background-color: red');
+        this.rerunBackgroundColor = 'background-color: red';
         this.rerunResult = '[Rerun failed]';
       }
```
Comment (MatthijsSmets) [inline src/app/report/edit-display/edit-display.component.ts:90]: When only using the boolean attribute you can refactor this method to only set the boolean
```ts
this.rerunSuccess = this.report == response
```

File: src/app/shared/components/toast/toast.component.ts [hunk 8]
```diff
@@ -1,22 +1,26 @@
...
   standalone: true,
-  imports: [NgFor, NgIf, NgbToast],
+  imports: [NgFor, NgIf, NgbToast, ClipboardModule],
 })
 export class ToastComponent implements OnInit, OnDestroy {
   toastSubscription!: Subscription;
   selectedAlert!: Toast;
   @ViewChild('modal') modal!: TemplateRef<Element>;
+  @ViewChild('detailedErrorMessage') detailedErrorMessageElement!: ElementRef;
+  @ViewChild('CopyToClipboard') copyToClipboardElement!: ElementRef<HTMLButtonElement>;
```
Comment (MatthijsSmets) [inline src/app/shared/components/toast/toast.component.ts:21]: These are not used, forgot to remove them?

**Review by MatthijsSmets — CHANGES_REQUESTED** [review]
Take a look at my comments, there's some stuff that can be refactored.

File: src/app/report/edit-display/edit-display.component.ts (hunk 7 above)
Comment (MarkvdVorst) [reply src/app/report/edit-display/edit-display.component.ts]: changed it to the suggested line

File: src/app/shared/components/toast/toast.component.ts (hunk 8 above)
Comment (MarkvdVorst) [reply src/app/shared/components/toast/toast.component.ts]: Yes I forgot to remove them. These two are gone now.

**Review by philipsens — APPROVED** [review]

**Review by Matthbo — DISMISSED** [review]

**Comment by MatthijsSmets** [comment]
> Klaar nu 🙄

I was not given a chance to review changes made by my request, I would appreciate it if you didn't close a PR if anyone still has an oustanding request changes.

**Pull request merged**